ui/pages/asset/asset.tsx

```diff
@@ -3,26 +3,43 @@ import { useSelector } from 'react-redux';
 import { Redirect, useParams } from 'react-router-dom';
 import { isEqualCaseInsensitive } from '../../../shared/modules/string-utils';
 import NftDetails from '../../components/app/assets/nfts/nft-details/nft-details';
-import {
-  getNativeCurrency,
-  getNfts,
-  getTokens,
-} from '../../ducks/metamask/metamask';
+import { getSelectedAccountTokensAcrossChains } from '../../selectors';
+import { getNativeCurrency, getNfts } from '../../ducks/metamask/metamask';
 import { DEFAULT_ROUTE } from '../../helpers/constants/routes';
 
 import NativeAsset from './components/native-asset';
 import TokenAsset from './components/token-asset';
 
+function findAssetByAddress(data: any, address?: string) {
+  if (!address) {
+    return null;
+  }
+
+  for (const chainId in data) {
+    if (chainId in data) {
```

Comment:
Is this `if` necessary?

Reply:
Refactored here: https://github.com/MetaMask/metamask-extension/pull/28193/commits/3b3ff1fa2c86d24da1a42c1b52af7d4d6fabd9e4